src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -533,22 +669,42 @@ internal static bool LocalFindAndPop(object callback)
                 return workItem;
             }
 
-            // Check for global work items
+            // Check for work items from the assigned global queue
+            if (AssignableWorkItemQueueCount > 0 && tl.assignedGlobalWorkItemQueue.TryDequeue(out workItem))
+            {
+                return workItem;
+            }
+
+            // Check for work items from the global queue
             if (workItems.TryDequeue(out workItem))
             {
                 return workItem;
             }
 
+            // Check for work items in other assignable global queues
+            uint randomValue = tl.random.NextUInt32();
+            if (AssignableWorkItemQueueCount > 0)
+            {
+                int queueIndex = tl.queueIndex;
+                for (int c = AssignableWorkItemQueueCount, maxIndex = c - 1, i = (int)(randomValue % (uint)c);
+                    c > 0;
+                    i = i < maxIndex ? i + 1 : 0, c--)
+                {
+                    if (i != queueIndex && assignableWorkItemQueues[i].TryDequeue(out workItem))
+                    {
+                        return workItem;
+                    }
+                }
+            }
```

Comment:
Interesting.  So every thread now not only checks it own local queue and the global queue and every other thread's local queue, but it also checks every global queue?  Might this not make the burst scenario have even more overhead, as all the threads will end up looking in even more places for work?

Reply:
Yes each thread would check a few more queues. On the 80-proc machine in addition to checking ~80 or more other threads' local queues, it would check 5 other global queues, so it may only be a small increase in overhead. The issue is covered by https://github.com/dotnet/runtime/issues/10752 and hasn't been fixed yet. Perhaps a way to fix it would be to limit the number of threads that do work-stealing at any given time.